flink-connectors/flink-connector-files/src/test/java/org/apache/flink/connector/file/src/FileSourceTextLinesITCase.java

```diff
@@ -61,20 +82,83 @@
 	@ClassRule
 	public static final TemporaryFolder TMP_FOLDER = new TemporaryFolder();
 
-	@ClassRule
-	public static final MiniClusterWithClientResource MINI_CLUSTER = new MiniClusterWithClientResource(
-		new MiniClusterResourceConfiguration.Builder()
-			.setNumberTaskManagers(1)
-			.setNumberSlotsPerTaskManager(PARALLELISM)
-			.build());
+	private static TestingMiniCluster miniCluster;
 
-	// ------------------------------------------------------------------------
+	private static TestingEmbeddedHaServices highAvailabilityServices;
+
+	private static CompletedCheckpointStore checkpointStore;
+
+	@BeforeClass
+	public static void setupClass() throws Exception  {
+		highAvailabilityServices = new HaServices(TestingUtils.defaultExecutor(),
+			() -> checkpointStore,
+			new StandaloneCheckpointIDCounter());
+
+		final Configuration configuration = createConfiguration();
+
+		miniCluster = new TestingMiniCluster(
+			new TestingMiniClusterConfiguration.Builder()
+				.setConfiguration(configuration)
+				.setNumTaskManagers(1)
+				.setNumSlotsPerTaskManager(PARALLELISM)
+				.setRpcServiceSharing(RpcServiceSharing.DEDICATED)
+				.build(),
+			() -> highAvailabilityServices);
+
+		miniCluster.start();
+	}
+
+	private static Configuration createConfiguration() throws IOException {
+		final Configuration configuration = new Configuration();
+		configuration.set(CheckpointingOptions.STATE_BACKEND, "filesystem");
```

Comment:
Does it make a difference here which state backend we use, "filesystem" versus the default memory state backend?
Otherwise I would leave this out of here, following the principle that a test should not deal with aspects unrelated to what it actually tests.

Reply:
I think, we can remove `filesystem`. I think it is just a leftover from my experiments that the checkpoint data stays available between fail-overs.